### ISSUE TITLE
Make VS Code less chatty with screen readers

### ISSUE THREAD

**Issue by MarcoZehe** [microsoft/vscode#114892]
Right now, several parts of the editor are equipped with landmark roles, like "complementary", "navigation", etc. These landmarks are traditionally associated with document-like structure, not with application interfaces. In addition, with several side bars open etc., the screen reader can get very chatty with saying "complementary" (five syllables) several dozen or hundred times per day.

My suggestion, to make VS Code feel more like a desktop app, would be to change some of these roles to be more widget-like in nature. For example:

- The navigation part, Parts.ACTIVITYBAR_PART, is merely a container for two toolbars. The toolbars themselves already announce themselves to screen readers, so the hierarchy doesn't really need to include the navigation part. Suggest to change this to either "group" or even "none" because it appears to be there mostly for presentation purposes.
- Parts.SIDEBAR_PART is currently a complementary, one of two, to be exact. The other is Parts.PANEL_PART. Both of these are merely groups that hold a tab list and corresponding tab panels. Again, "group" or even make them disappear via role "none" might be a better fit.
- I am not yet sure what to do with the  Parts.TITLEBAR_PART, it is currently a "contentinfo", which kinda makes sense, but is again a pure landmark role.
- Since "Main" is short enough, and kind of is the right thing to say anyway, I would leave that unchanged.

I'm putting this up for discussion here since I realize that some people might disagree with me on all or some of these points. But I wanted to bring it up anyway since right now, with these landmark roles, VS Code literally screams at screen readers "Hey, at heart, I am actually a website!". Or at least that's how I feel like when using it.

CC @isidorn , @leonardder, @joanmarie, @marlon-sousa, @jvesouza.

**Comment by jvesouza** [comment]
@MarcoZehe Thanks for report this issue.
I wanted to report something similar to what you reported but I was unable to elaborate the problem with this wealth of details.

A problem that I have encountered frequently when editing java code is that when the edited class has errors and I try to use the auto complete function, the error is shown before the suggestions.
This decreases productivity because I have to wait for the screen reader to finish reading the error before I can hear the suggestions.

**Comment by leonardder** [comment]
@MarcoZehe Have you considered disabling NVDA's landmark reporting just for Code as a workaround?
Note that this of course partially relates to that NVDA reports landmarks in focus mode, which might be a point of debate. However, I think it makes sense that NVDA behaves this way.

**Comment by isidorn** [comment]
@MarcoZehe thanks a lot for providing feedback.
Even though @leonardder has provided a workaround I think we should behave nicely out of the box. 
I suggest that I change the role of `ACTIVITYBAR_PART`, `SIDEBAR_PART` and `PANEL_PART` to have a `role`: `none`.
I plan to do this start of next milestone (so in 2 weeks in Insiders) then you can try it out and let me know what you think. 

After that we can do some more fine tuning. Let me know what you think @MarcoZehe 

@jvesouza that sounds like an unrelated issue to this, maybe you file a new issue and we can continue the discussion there? Sounds like the Errors get read before suggestions which does not make sense...

**Comment by MarcoZehe** [comment]
> @MarcoZehe Have you considered disabling NVDA's landmark reporting just for Code as a workaround?

No, since I currently work primarily on a Mac. I could try this with VoiceOver for a VS Code specific activity, but thought that this warrants a discussion for default behavior since we talk about VS Code being a desktop app and not a website. And while you can distribute this default setting in a VS Code module for NVDA, and possibly scripts for JAWS if Vispero were to include that, providing a VS Code specific activity for VoiceOver users is not possible easily.

**Comment by MarcoZehe** [comment]
> Even though @leonardder has provided a workaround I think we should behave nicely out of the box.

👍🏼

> I suggest that I change the role of `ACTIVITYBAR_PART`, `SIDEBAR_PART` and `PANEL_PART` to have a `role`: `none`.
> I plan to do this start of next milestone (so in 2 weeks in Insiders) then you can try it out and let me know what you think.

This sounds awesome, thank you!

**Comment by jvesouza** [comment]
> @jvesouza that sounds like an unrelated issue to this, maybe you file a new issue and we can continue the discussion there? Sounds like the Errors get read before suggestions which does not make sense...

@isidorn After I restarted my machine the problem seems to be gone, although I still find it eventually.
I need to investigate further to try to reproduce.

**Comment by isidorn** [comment]
@jvesouza okey, let me know if you find steps!

**Comment by isidorn** [comment]
@MarcoZehe as we agreed upon I have changed the `ACTIVITYBAR_PART`,` SIDEBAR_PART` and `PANEL_PART` to have a `role: none`.
Reading the aria docs, that should not be the same as not setting any role on them at all. So I hope this is the correct way to tackle this.

Anyways try it out with Insiders from Friday and let me know how it goes.
Thanks a lot for the feedback and keep it coming!

**Comment by MarcoZehe** [comment]
@isidorn I just received this Insider build, and I must say I like the change. It definitely streamlines the experience for me. What do you think @leonardder @jvesouza @zersiax @neurrone

**Comment by leonardder** [comment]
It's indeed much less chatty now.

**Comment by isidorn** [comment]
Lovely, let's stick with this change for now.
Hopefully this will also make sense for new VS Code users.

**Comment by Neurrone** [comment]
👍 to this change as well

**Comment by jvesouza** [comment]
In my opinion it was great.
